### PR TITLE
a basic recipe can be added to DataStore

### PR DESCRIPTION
**Pull request by isamitre**

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/NewRecipeServlet.java [hunk 1]
```diff
@@ -27,9 +17,39 @@
...
+    String recipeName = request.getParameter("recipeName");
+    String tagsResponse = request.getParameter("tags");
+    String ingredientsResponse = request.getParameter("ingredients");
+    String stepsResponse = request.getParameter("steps");
+    String privacy = request.getParameter("privacy");
 
-    response.sendRedirect("/pages/UserPage.html");
-  }
+    List<String> tags = new ArrayList<String>(Arrays.asList(tagsResponse.split(",", 0)));
+    List<String> ingredients = new ArrayList<String>(Arrays.asList(ingredientsResponse.split("\n", 0)));
```
Comment (ez-may) [inline src/main/java/com/google/sps/servlets/NewRecipeServlet.java]: Some comments here and below to explain what you're doing and why would help in the future.

File: src/main/java/com/google/sps/servlets/NewRecipeServlet.java [hunk 2]
```diff
@@ -27,9 +17,39 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // implement new recipe
+    String recipeName = request.getParameter("recipeName");
+    String tagsResponse = request.getParameter("tags");
+    String ingredientsResponse = request.getParameter("ingredients");
+    String stepsResponse = request.getParameter("steps");
+    String privacy = request.getParameter("privacy");
```
Comment (ez-may) [inline src/main/java/com/google/sps/servlets/NewRecipeServlet.java:25]: It doesn't look like you use this anywhere, but I'm assuming that's later in the implementation?

File: src/main/java/com/google/sps/servlets/NewRecipeServlet.java [hunk 3]
```diff
@@ -27,9 +17,39 @@
...
+    }
+    for (String ingredient : ingredients) {
+      ingredient.trim();
+    }
+    for (String step : steps) {
+      step.trim();
+    }
 
-}
+    Entity recipeEntity = new Entity("PrivateRecipe");
```
Comment (ez-may) [inline src/main/java/com/google/sps/servlets/NewRecipeServlet.java:37]: Will all recipes be saved as this type of entity later, or will you include an if-statement later to control the type of entity creation?

File: src/main/webapp/pages/UserPage.html [hunk 4]
```diff
@@ -47,8 +47,8 @@ <h1 class="center">Create Recipe</h1>
             <!-- <option value="public">public</option> -->
             </select>
             <br><br>
-            <label for="image">Upload an image</label>
-            <input type="file"  accept="image/*" name="image" id="image">
+            <!-- <label for="image">Upload an image</label>
```
Comment (ez-may) [inline src/main/webapp/pages/UserPage.html:51]: Are you leaving this to keep working on it later?

File: src/main/java/com/google/sps/servlets/NewRecipeServlet.java [hunk 5]
```diff
@@ -27,9 +17,39 @@
...
+    String privacy = request.getParameter("privacy");
 
-    response.sendRedirect("/pages/UserPage.html");
-  }
+    List<String> tags = new ArrayList<String>(Arrays.asList(tagsResponse.split(",", 0)));
+    List<String> ingredients = new ArrayList<String>(Arrays.asList(ingredientsResponse.split("\n", 0)));
+    List<String> steps = new ArrayList<String>(Arrays.asList(stepsResponse.split("\n", 0)));
+
+    tags.removeAll(Arrays.asList("", null));
+    ingredients.removeAll(Arrays.asList("", null, "\n", "\r\n", "\r"));
```
Comment (ez-may) [inline src/main/java/com/google/sps/servlets/NewRecipeServlet.java:34]: A comment here would make this understandable.

File: src/main/java/com/google/sps/servlets/NewRecipeServlet.java (hunk 3 above)
Comment (isamitre) [reply src/main/java/com/google/sps/servlets/NewRecipeServlet.java:37]: later when we have public recipes I'll have an if statement

File: src/main/java/com/google/sps/servlets/NewRecipeServlet.java (hunk 2 above)
Comment (isamitre) [reply src/main/java/com/google/sps/servlets/NewRecipeServlet.java:25]: yup once we have public recipes

File: src/main/webapp/pages/UserPage.html (hunk 4 above)
Comment (isamitre) [reply src/main/webapp/pages/UserPage.html:51]: yeah I tried to make it work but it's really not lol so for the MVP I'll leave it and work on it again later

Comment (ez-may) [reply src/main/webapp/pages/UserPage.html:51]: Ok, then I would say add a comment with a to-do, and maybe the reason it's not MVP.

**Review by ez-may — APPROVED** [review]
LGTM

File: src/main/java/com/google/sps/servlets/NewRecipeServlet.java [hunk 6]
```diff
@@ -27,9 +17,41 @@
...
+
+    for (String tag : tags) {
+      tag.trim();
+    }
+    for (String ingredient : ingredients) {
+      ingredient.trim();
+    }
+    for (String step : steps) {
+      step.trim();
+    }
```
Comment (oturner24) [inline src/main/java/com/google/sps/servlets/NewRecipeServlet.java]: I think you could do this as part of the splitting step above, and avoid iterating over all the lists. Here's a stackoverflow: https://stackoverflow.com/questions/41953388/java-split-and-trim-in-one-shot. You could do something like:
`List<String> tags = new ArrayList<String>(Arrays.asList(tagsResponse.trim().split("\\s*,\\s*");`
where you trim the original string, and then include whitespace in the split by using regular expressions. (if you're not familiar with regular expressions, here's a resource: https://medium.com/factory-mind/regex-tutorial-a-simple-cheatsheet-by-examples-649dc1c3f285. They're super handy for string parsing)

Or if you don't want to do that, take a look at Java 8 streams (see the second answer in that stackoverflow). It'll still be iterating over the lists but with much more concise (and in my opinion, more readable) syntax.

**Review by oturner24 — APPROVED** [review]

**Pull request merged**